publication-rest/src/test/java/no/unit/nva/publication/fetch/FetchImportCandidateHandlerTest.java

```diff
@@ -0,0 +1,151 @@
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.github.tomakehurst.wiremock.junit5.WireMockTest;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import no.unit.nva.identifiers.SortableIdentifier;
+import no.unit.nva.model.AdditionalIdentifier;
+import no.unit.nva.model.Contributor;
+import no.unit.nva.model.EntityDescription;
+import no.unit.nva.model.Identity;
+import no.unit.nva.model.Organization;
+import no.unit.nva.model.PublicationDate;
+import no.unit.nva.model.ResearchProject;
+import no.unit.nva.model.ResourceOwner;
+import no.unit.nva.model.Username;
+import no.unit.nva.model.funding.FundingBuilder;
+import no.unit.nva.model.role.Role;
+import no.unit.nva.model.role.RoleType;
+import no.unit.nva.publication.model.business.ImportCandidate;
+import no.unit.nva.publication.model.business.ImportStatus;
+import no.unit.nva.publication.service.ResourcesLocalTest;
+import no.unit.nva.publication.service.impl.ResourceService;
+import no.unit.nva.testutils.HandlerRequestBuilder;
+import nva.commons.apigateway.GatewayResponse;
+import nva.commons.apigateway.exceptions.NotFoundException;
+import nva.commons.core.Environment;
+import org.apache.http.entity.ContentType;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.zalando.problem.Problem;
+
+@WireMockTest(httpsEnabled = true)
+public class FetchImportCandidateHandlerTest extends ResourcesLocalTest {
+
+    public static final String IDENTIFIER = "importCandidateIdentifier";
+    private ByteArrayOutputStream output;
+    private Context context;
+    private ResourceService resourceService;
+    private FetchImportCandidateHandler handler;
+
+    @BeforeEach
+    public void setUp() {
+        super.init();
+        Environment environment = mock(Environment.class);
+        when(environment.readEnv(ALLOWED_ORIGIN_ENV)).thenReturn("*");
+        when(environment.readEnv(ENV_NAME_NVA_FRONTEND_DOMAIN)).thenReturn("localhost");
+        resourceService = new ResourceService(client, Clock.systemDefaultZone());
+        context = mock(Context.class);
+        output = new ByteArrayOutputStream();
+        handler = new FetchImportCandidateHandler(resourceService);
+    }
+
+    @Test
+    void shouldReturnImportCandidateSuccessfully() throws NotFoundException, IOException {
+        var importCandidate = createPersistedImportCandidate();
+        var request = createRequest(importCandidate.getIdentifier());
+        handler.handleRequest(request, output, context);
+        var response = GatewayResponse.fromOutputStream(output, ImportCandidate.class);
+        var responseImportCandidate = response.getBodyObject(ImportCandidate.class);
+
+        assertThat(importCandidate.getIdentifier(), is(equalTo(responseImportCandidate.getIdentifier())));
```

Comment:
Why isn't assertThat(responseImportCandidate, is(equalTo(importCandidate)) possible?

Also I think you have flipped the order of expected and actual object.

Atleast check if the import status is present; it's the only thing that differ from a regular resource.